src/main/kotlin/no/nav/familie/ef/sak/integration/JournalpostClient.kt

```diff
@@ -24,19 +26,24 @@ class JournalpostClient(@Qualifier("azure") restOperations: RestOperations,
 
     fun hentJournalpost(journalpostId: String): Journalpost {
         return getForEntity<Ressurs<Journalpost>>(URI.create("${journalpostURI}?journalpostId=${journalpostId}")).data
-               ?: error("Kunne ikke hente journalpost med id ${journalpostId}")
+                ?: error("Kunne ikke hente journalpost med id ${journalpostId}")
     }
 
-    fun hentDokument(journalpostId: String, dokumentInfoId: String): ByteArray {
-        return getForEntity<Ressurs<ByteArray>>(URI.create("${journalpostURI}/hentdokument/${journalpostId}/${dokumentInfoId}")).data
-               ?: error("Kunne ikke hente dokument for journalpost=${journalpostId} og dokumentInfoId=${dokumentInfoId}")
+    fun hentDokument(journalpostId: String, dokumentInfoId: String, dokumentVariantformat: DokumentVariantformat): ByteArray {
+        return getForEntity<Ressurs<ByteArray>>(UriComponentsBuilder
+                .fromUriString("${journalpostURI}/hentdokument/${journalpostId}/${dokumentInfoId}")
+                .queryParam("variantFormat", dokumentVariantformat)
+                .build()
+                .toUri())
+                .data
```

Comment:
Jeg tror vi kan utgå fra att hvis ressurs har data så er den suksess, men tenker nesten vi burde sjekke på status her?

Reply:
getForEntity har allerede sjekket at status er 200 her, så det burde ikke være nødvendig.

Reply:
kan kanskje bruke getDataOrThrow i stedet for data. Da får du error med feilmeldingen fra ressurs. Tror det i de fleste tilfeller er mer nyttig.